### PR TITLE
First version of powsybl-eurostag

### PR DESCRIPTION
**Pull request by mathbagu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: eurostag-model/src/main/java/com/powsybl/eurostag/model/EsgGenerator.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    private final double pgmin; // minimum active power [MW]
+    private final double pgen; // active power [MW]
+    private final double pgmax; // maximum active power [MW]
+    private final double qgmin; // minimum reactive power [Mvar]
+    private final double qgen; // reactive power [Mvar]
+    private final double qgmax; // maximum reactive power [Mvar]
+    private EsgRegulatingMode xregge; // regulating mode
+    private double vregge; // voltage target
+    private final Esg8charName zregnoge; // regulated node (= ZNODGE if blank)
+    private final double qgensh; // Reactive sharing coefficient [%]
```
Comment (annetill) [inline eurostag-model/src/main/java/com/powsybl/eurostag/model/EsgGenerator.java:31]: I think that we have to use the extension `CoordinatedReactiveControl` to store the reactive sharing coefficient. It seems that we don't do that in the convserion.

File: eurostag-converter/src/main/java/com/powsybl/eurostag/converter/EurostagEchExport.java [hunk 2]
```diff
@@ -0,0 +1,938 @@
...
+
+            //**************************
+            //*** LOSSES COMPUTATION *** (Record 1)
+            //**************************
+
+            double nomiU2 = twt.getTerminal2().getVoltageLevel().getNominalV();
+
+            //...mTrans.getR() = Get the nominal series resistance specified in Ω at the secondary voltage side.
+            double rpu2 = (twt.getR() * parameters.getSnref()) / nomiU2 / nomiU2;  //...total line resistance  [p.u.](Base snref)
+            double gpu2 = ((config.isSpecificCompatibility() ? twt.getG() / 2.0 : twt.getG()) / parameters.getSnref()) * nomiU2 * nomiU2;  //...semi shunt conductance [p.u.](Base snref)
```
Comment (annetill) [inline eurostag-converter/src/main/java/com/powsybl/eurostag/converter/EurostagEchExport.java:435]: Do we need to use the `specificCompatibility` parameter provided by the `LoadFlowParameters`?

File: eurostag-model/src/main/java/com/powsybl/eurostag/model/EsgGenerator.java (hunk 1 above)
Comment (mathbagu) [reply eurostag-model/src/main/java/com/powsybl/eurostag/model/EsgGenerator.java:31]: When the first version of this exporter has been implemented, the extension you are talking about didn't exist. I agree with your proposal: do you think we should do it in this PR or create an issue and do it later?

Comment (annetill) [reply eurostag-model/src/main/java/com/powsybl/eurostag/model/EsgGenerator.java:31]: I will find time to make it in that PR.

**Comment by annetill** [comment]
I have compared the voltages after an open-loadflow calculation and an Eurostag calculation. It is similar. We can merge this PR as a first working version of the exporter.

**Review by annetill — APPROVED** [review]
Thanks a lot for your last fix ;-)

**Pull request merged**